next.dynamic.mjs

```diff
@@ -212,6 +212,21 @@ const getDynamicRouter = async () => {
       path
     );
 
+    const match = path.match(/blog\/(release|vulnerability)/);
+
+    if (match) {
+      const category = match[1];
+      const currentFile = siteConfig.rssFeeds.find(
+        item => item.category === category
+      )?.file;
+      // Dynamically construct the XML path for blog/release and blog/vulnerability
+      pageMetadata.alternates.types['application/rss+xml'] =
+        `${baseUrlAndPath}/en/feed/${currentFile}`;
+    } else {
+      pageMetadata.alternates.types['application/rss+xml'] =
+        `${baseUrlAndPath}/en/feed/blog.xml`;
+    }
```

Comment:
```suggestion
    pageMetadata.alternates.types['application/rss+xml'] =
      `${baseUrlAndPath}/en/feed/blog.xml`;
     
    // Dynamically construct the XML path for blog/release and blog/vulnerability   
    const blogMatch = path.match(/blog\/(release|vulnerability)/);
    if (blogMatch) {
      const category = blogMatch[1];
      const currentFile = siteConfig.rssFeeds.find(
        item => item.category === category
      )?.file;
      if (currentFile) {
        pageMetadata.alternates.types['application/rss+xml'] =
          `${baseUrlAndPath}/en/feed/${currentFile}`;
      }
    }
```
I'm curious what @ovflowd thinks here, but I wonder if we'd be better off defaulting to always setting the RSS feed to the blog, and then overwriting it if we find something "special"? With the current logic, it seems there is a risk that `currentFile` might be undefined (though in reality it won't as we know that `siteConfig.rssFeeds` has what we need).

Also, very much a nit, but `match` => `blogMatch` as a better variable name to make it clearer what it is.

Reply:
Yeah I don't see why it will be undefined, after we update the regex we can make sure that it matches with the path properly

Reply:
Done